### PR TITLE
Adds PublishNotificationAction for legacy ISM implementation

### PR DESCRIPTION
**Pull request by dbbaughe**
Signed-off-by: Drew Baugher <46505179+dbbaughe@users.noreply.github.com>

### Description
Adds a publish notification action as a pass through API for the ISM plugin.
It is an internal only API meant only for ISM and is not used in REST layer.

### Issues Resolved
https://github.com/opensearch-project/notifications/issues/253

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbbaughe** [comment]
Unresolved references etc. failures -> waiting on this PR to be merged first in common utils
https://github.com/opensearch-project/common-utils/pull/45

**Comment by zhongnansu** [comment]
Could you update your PR to pass the CI? The upstream CI failure was temporarily fixed. So PRs should pass CI now

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt [hunk 1]
```diff
@@ -204,6 +224,51 @@ object SendMessageActionHelper {
...
+     * refactor this to have it's own internal private spi call to completely decouple them instead.
+     *
+     * @param baseMessage legacy base message
+     * @return notification delivery status for the legacy destination
+     */
+    private fun sendMessageToLegacyDestination(baseMessage: LegacyBaseMessage): LegacyDestinationResponse {
+        val message = MessageContent(title = "Index Management Notification", textDescription = baseMessage.messageContent)
+        // These legacy destination calls do not have reference Ids, just passing index management feature constant
+        return when (baseMessage.channelType) {
+            LegacyDestinationType.LEGACY_SLACK -> {
```
Comment (zhongnansu) [inline notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt:242]: Np: But I wonder why only slack/chime/webhook are added here?

**Review by zhongnansu — APPROVED** [review]

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt (hunk 1 above)
Comment (dbbaughe) [reply notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt:242]: These are the only supported destination types that existed in the legacy implementation in ISM.

**Pull request merged**